### PR TITLE
mender-monitor_git: Beatifications and robustness when generating PV

### PR DESCRIPTION
**Pull request by lluiscampos**
Previously, build from tarball would generate:
* `mender-monitor-master-gitae0ead3-r0` for internal builds (CI)
* `mender-monitor-master-git-r0` for external builds (users)

This commit makes it a bit nicer:
* `mender-monitor-master-git+ae0ead3-r0` for internal builds
* `mender-monitor-master-r0`for external builds

Also made the parsing more robust, so that it explicitly fails when the
filename is unknown.

Original work by Alan Martinovic <alan.martinovic@northern.tech>

### CODE REVIEW

**Comment by lluiscampos** [comment]
@mender-test-bot start pipeline :classical_building:

File: meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor_git.bb [hunk 1]
```diff
@@ -17,16 +19,28 @@ def mender_monitor_srcrev_from_src_uri(d, src_uri):
...
+        if filename == "mender-monitor-master.tar.gz":
+            # Building from external tarball, do not append git info
+            return ""
+        m = re.match(r"mender-monitor-([0-9]+\.[0-9]+\.[0-9]+(?:-build[0-9]+)?)\.tar\.gz", filename)
+        if m is not None:
+            # The tarball is a tagged version, in which case SRCREV is not to
+            # be used but still needs to parse
             return ""
-        return m.group(1)
+        m = re.match(r"mender-monitor-([0-9a-f]{7})\.tar\.gz", filename)
```
Comment (kacf) [inline meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor_git.bb:37]: I think that Git uses more characters once the repository reaches a certain size (run `git rev-parse --short HEAD` in the gui repo for an example). Not completely sure if that makes its way here, but maybe it's better to use `{7,}`?

**Review by kacf — APPROVED** [review]

File: meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor_git.bb (hunk 1 above)
Comment (lluiscampos) [reply meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor_git.bb:37]: I didn't know about this! Makes sense. I'll amend.

**Comment by lluiscampos** [comment]
> @mender-test-bot start pipeline classical_building

:repeat_one:

**Pull request merged**